contracts/options.cairo

```diff
@@ -516,30 +545,34 @@ func _burn_option_token_long{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, ran
         //        min(size of pools short, amount_in_pool_currency)
         //        since the pools' short might not be covering all of the long
 
-        let (current_locked_balance) = pool_locked_capital.read(lptoken_address);
+        let (current_locked_balance) = get_pool_locked_capital(lptoken_address);
         let (size_to_be_unlocked_in_base) = min(pool_short_position, option_size);
-        let (size_to_be_unlocked) = convert_amount_to_option_currency_from_base(
-            size_to_be_unlocked_in_base, option_type, strike_price
+
+        let size_to_be_unlocked_in_base_uint256 = intToUint256(size_to_be_unlocked_in_base);
+        let strike_price_uint256 = toUint256_balance(strike_price, quote_address);
+        let (size_to_be_unlocked: Uint256) = convert_amount_to_option_currency_from_base_uint256(
+            size_to_be_unlocked_in_base_uint256, option_type, strike_price_uint256, base_address
         );
-        let new_locked_balance = Math64x61.sub(current_locked_balance, size_to_be_unlocked);
-        pool_locked_capital.write(lptoken_address, new_locked_balance);
+        let (new_locked_balance: Uint256) = uint256_sub(current_locked_balance, size_to_be_unlocked);
+        set_pool_locked_capital(lptoken_address, new_locked_balance);
 
         // Update pool's short position
-        let (pools_short_position) = option_position.read(
+        let (pools_short_position) = get_option_position(
             lptoken_address, TRADE_SIDE_SHORT, maturity, strike_price
         );
-        let new_pools_short_position = Math64x61.sub(pools_short_position, size_to_be_unlocked_in_base);
-        option_position.write(
+        let pools_short_position_uint256 = intToUint256(pools_short_position);
+        let new_pools_short_position = pools_short_position - size_to_be_unlocked_in_base;
```

Comment:
btw, is there something like safe math, or something like this for felts?

all over this module there are `+` and `-` but we don't check for overflowing

Reply:
Fixed in 577c8dd by adding assert_nn(new_pools_long_position); where it might be an issue.